### ISSUE TITLE
Nanopub templates: support datatype literals

### ISSUE THREAD

**Issue by Ostrzyciel** [knowledgepixels/nanodash#152]
I may be wrong here, but as far as I understand, there is currently no way to specify a placeholder in a Nanopub template that would correspond to a datatype literal (e.g., a literal with type xsd:int). When I use the "Literal placeholder" type and put (for example) `"2024-08-14"^^xsd:date` in the template, the entire thing (including the quotes and the datatype annotation) is used as the lexical value of the literal. When I use the "URI/literal placeholder" type and do the same, I get an "invalid literal value" error:

![image](https://github.com/user-attachments/assets/00d37a19-9f6c-491d-85aa-782e8b789861)
![image](https://github.com/user-attachments/assets/9e45f396-eede-41be-9473-5d233e8a15e7)

So, I have no way of obtaining a triple like `_:s _:p "2024-08-14"^^xsd:date`.

What I think would be perfect for this:

- Have a new placeholder type: `DatatypeLiteralPlaceholder`, a subclass of [`LiteralPlaceholder`](https://peta-pico.github.io/npo/ntemplate/index-en.html#https://w3id.org/np/o/ntemplate/LiteralPlaceholder)
- Specify the datatype for the placeholder in the Nanopub template (e.g., xsd:date). For this probably a new property in the template vocabulary would be needed.
- Optional: in Nanodash's validation code implement checks for common datatypes (e.g., xsd:date, xsd:int) to make sure the values inputted by the user are valid for this datatype. This theoretically could also be done with `hasRegex`, but then each template creator would have to come up with their own regexes for datatypes (assuming one could even express this using a regex... well, we don't need to be perfect).

Mockup of how the user would see this:
![image](https://github.com/user-attachments/assets/2bec0a70-d028-4638-b86c-911a78d7e09a)

Note that this proposal is based on the understanding that there is no other way to have datatype literals in Nanodash.

**Comment by tkuhn** [comment]
Yes, exactly, that is what I plan to do, but it's not yet implemented. There is this other ticket about this too: https://github.com/knowledgepixels/nanodash/issues/14

It would then also be good to have a nice date picker for dates, and possibly other GUI elements for the other datatypes.

Is this something you would need urgently? I cannot promise too much, but I can give it higher priority.

**Comment by Ostrzyciel** [comment]
Thank you for the quick reply!

Not really needed urgently... I can kind of manage without it for now.
